tests/unit/test_handler.py

```diff
@@ -1,73 +1,178 @@
-            "Host": "1234567890.execute-api.us-east-1.amazonaws.com",
-            "X-Forwarded-Proto": "https",
-            "X-Amz-Cf-Id": "aaaaaaaaaae3VYQb9jd-nvCd-de396Uhbp027Y2JvkCPNLmGJHqlaA==",
-            "CloudFront-Is-Tablet-Viewer": "false",
-            "Cache-Control": "max-age=0",
-            "User-Agent": "Custom User Agent String",
-            "CloudFront-Forwarded-Proto": "https",
-            "Accept-Encoding": "gzip, deflate, sdch",
-        },
-        "pathParameters": {"proxy": "/examplepath"},
-        "httpMethod": "POST",
-        "stageVariables": {"baz": "qux"},
-        "path": "/examplepath",
+mock_ce_response = {
+  'GroupDefinitions': [
+    {'Type': 'COST_CATEGORY', 'Key': 'Owner Email'}
+  ],
+  'ResultsByTime': [
+    {
+      'TimePeriod': {'Start': '2023-03-01', 'End': '2023-04-01'},
+       'Total': {},
+      'Groups': [
+          {
+            'Keys': ['Owner Email$'],
+            'Metrics': {
+              'UnblendedCost': {
+                'Amount': '48446.9515396549',
+                'Unit': 'USD'
+              }
+            }
+          },
+          {'Keys': ['Owner Email$user1@synapse.org'], 'Metrics': {'UnblendedCost': {'Amount': '0.30000000264', 'Unit': 'USD'}}},
+          {'Keys': ['Owner Email$user2@synapse.org'], 'Metrics': {'UnblendedCost': {'Amount': '3.000000264', 'Unit': 'USD'}}},
+          {'Keys': ['Owner Email$user3@sagebase.org'], 'Metrics': {'UnblendedCost': {'Amount': '30.000000264', 'Unit': 'USD'}}},
+        ],
+        'Estimated': False
     }
+  ],
+  'DimensionValueAttributes': [],
+  'ResponseMetadata': {
+    'RequestId': '15f5cdde-0ea0-420f-867a-b862afeca967',
+    'HTTPStatusCode': 200,
+    'HTTPHeaders': {
+      'date': 'Wed, 31 May 2023 20:57:36 GMT',
+      'content-type': 'application/x-amz-json-1.1',
+      'content-length': '14561',
+      'connection': 'keep-alive',
+      'x-amzn-requestid': '15f5cdde-0ea0-420f-867a-b862afeca967',
+      'cache-control': 'no-cache'
+    },
+    'RetryAttempts': 0
+  }
+}
+
+expected_ce_costinfo = [
+ {'Estimated': False,
+  'Groups': [{'Keys': ['Owner Email$'],
+              'Metrics': {'UnblendedCost': {'Amount': '48446.9515396549',
+                                            'Unit': 'USD'}}},
+             {'Keys': ['Owner Email$user1@synapse.org'],
+              'Metrics': {'UnblendedCost': {'Amount': '0.30000000264',
+                                            'Unit': 'USD'}}},
+             {'Keys': ['Owner Email$user2@synapse.org'],
+              'Metrics': {'UnblendedCost': {'Amount': '3.000000264',
+                                            'Unit': 'USD'}}},
+             {'Keys': ['Owner Email$user3@sagebase.org'],
+              'Metrics': {'UnblendedCost': {'Amount': '30.000000264',
+                                            'Unit': 'USD'}}}],
+  'TimePeriod': {'End': '2023-04-01',
+                 'Start': '2023-03-01'},
+  'Total': {}},
+]
+
+expected_emails_all = {
+  '': 48446.9515396549,
```

Comment:
why is this an empty key?  what is this number?

Reply:
great question, this represents the total for uncategorized costs, so in our case thats the total for resources without an `OwnerEmail` or `synapse:email` tag

Reply:
would it make sense to set the key to `uncategorized` and also provide comments explaining this?